pkg/landscaper/controllers/execution/controller.go

```diff
@@ -114,6 +114,19 @@ func HandleAnnotationsAndGeneration(ctx context.Context, log logr.Logger, c clie
 		}
 		log.V(7).Info("successfully updated metadata")
 	}
+
+	// also reset the phase when the force reconcile annotation is present.
+	// Otherwise we would never bbe able to leave a final phase
+	if lsv1alpha1helper.HasOperation(exec.ObjectMeta, lsv1alpha1.ForceReconcileOperation) {
```

Comment:
This works. But why does this method handle the force reconcile different than the normal reconcile? Wouldn't it be simpler to include the force reconcile case in the condition in line 94?
```
if hasReconcileAnnotation || hasForceReconcileAnnotation || exec.Status.ObservedGeneration != exec.Generation {
```

Reply:
I also tried that first but there is a problem with your suggested approach.

The problem is the different annotation handling.
The reconcile annotation is directly removed in this function as soon as the status is set.
Which means that event with additional reconciles we do not reset the phase.

The force-reconcile annotation is handled differently as it may stay for multiple reconcile loops.
Which means that this function could be called multiple times within one force-reconcile loop which would lead to always update/reset the status.
Thats why we only want to update if the status is in a final state. To get it out of the final state but not update progressing deploy items